src/HtmlProcessor/AbstractHtmlProcessor.php

```diff
@@ -271,7 +271,8 @@ private function normalizeDocumentType(string $html): string
      */
     private function addContentTypeMetaTag(string $html): string
     {
-        $hasContentTypeMetaTag = \stripos($html, 'Content-Type') !== false;
+        $hasContentTypeMetaTag
+            = \preg_match('%<meta(?=\\s)[^>]*\\shttp-equiv=(["\']?+)Content-Type\\g{-1}[\\s/>]%i', $html) === 1;
```

Comment:
With the `=== 1`,  we've switched the meaning from "at least one content-type meta tag" to "exactly one content-type meta tag".

Even if multiple occurrences do not make much sense, I'd still prefer us to recognize this as "there is one" by keeping using `!== false`, and also cover this edge case with a test.

Reply:
I don't quite understand the `(?=\\s)` part yet. What does it do?

Reply:
> 
> 
> With the `=== 1`, we've switched the meaning from "at least one content-type meta tag" to "exactly one content-type meta tag".
> 
> Even if multiple occurrences do not make much sense, I'd still prefer us to recognize this as "there is one" by keeping using `!== false`, and also cover this edge case with a test.

The return value does appear to indicate the number of matches (`0` or `1` because it is limited to one match) with `false` on error.  So to be more explicit we'd probably need something like

```php
$contentTypeMetaTagMatchCount = \preg_match(...);
$hasContentTypeMetaTag = $contentTypeMetaTagMatchCount !== false && $contentTypeMetaTagMatchCount > 0;
```

Reply:
> 
> 
> I don't quite understand the `(?=\\s)` part yet. What does it do?

It asserts that at that position the next character is a whitespace but without 'consuming' it ('lookahead').  If it was 'consumed', `\\shttp-equiv` would not match if there was nothing else beetween `meta` and `http-equiv` because the whitespace to be mached there would have already been consumed.  Without asserting `meta` is followed by whitespace, other tag names beginning with `meta` (hitherto as yet not defined) would also be matched.

Reply:
> So to be more explicit we'd probably need something like
> 
> ```
> $contentTypeMetaTagMatchCount = \preg_match(...);
> $hasContentTypeMetaTag = $contentTypeMetaTagMatchCount !== false && $contentTypeMetaTagMatchCount > 0;
> ```

Or even more explicit:

```php
$hasContentTypeMetaTag = \is_int($contentTypeMetaTagMatchCount) false && $contentTypeMetaTagMatchCount > 0;
```

Reply:
Thanks for the explanation!